lib/AST/ASTPrinter.cpp

```diff
@@ -907,14 +907,11 @@ class PrintAST : public ASTVisitor<PrintAST> {
         DC = DC->getParent()->getInnermostTypeContext();
```

Comment:
using `Current->getInnermostDeclContext()` for the `mapTypeOutOfContext()` call sounds reasonable to me. 
However, using `Current->getDeclContext()` for `getMemberSubstitutions()` still has a problem when we have more  then two levels of nested type. I think there is really no way around checking whether dc is valid before calling `getMemberSubstitutions`.